### PR TITLE
feat: remove deprecation for the page slot

### LINKED ISSUE #7054: Remove deprecation in page slot component
Applies to:

- projects\storefrontlib\src\cms-structure\page\slot\page-slot.component.ts
```
   * @deprecated we'll stop supporting this property in 2.0 as
   * @deprecated since version 1.4
```

### PR DESCRIPTION
**Pull request by tobi-or-not-tobi**
close #7054 

BREAKING CHANGE:

The `PageSlotComponent` now requires new parameter `CmsConfig` and `ChangeDetectorRef`. The `CmsConfig` is used to read the configurable _page fold_ from the layout configuration. The `ChangeDetectorRef` is needed to update the slot _pending_ state asynchronously. The page fold and pending state have been added in release 1.4.

### CODE REVIEW

File: projects/storefrontlib/src/cms-structure/page/slot/page-slot.component.ts [hunk 1]
```diff
@@ -4,152 +4,151 @@ import {
...
    */
-  @Input()
-  set position(position: string) {
-    this.position$.next(position);
-    this.renderer.addClass(this.hostElement.nativeElement, position);
-  }
-  get position(): string {
-    return this.position$.value;
-  }
+  @Input() position: string;
```
Comment (dunqan) [inline projects/storefrontlib/src/cms-structure/page/slot/page-slot.component.ts]: I'm wondering if changing it to `string` and taking it into account only in `OnInit` is safe enough. 
Even if we agree that's it's safe in our codebase, the component is still part of a public API and I suspect more creative customers can come up with dynamic/changing position input quite easily. 
We can also state it's not supported but it's worth documenting then.

File: projects/storefrontlib/src/cms-structure/page/slot/page-slot.component.ts [hunk 2]
```diff
@@ -14,145 +14,184 @@ import {
...
-  @Input()
-  set position(position: string) {
-    this.position$.next(position);
-    this.renderer.addClass(this.hostElement.nativeElement, position);
+  @Input() set position(value: string) {
+    this.position$.next(value);
   }
   get position(): string {
-    return this.position$.value;
+    return this.position$?.value;
```
Comment (dunqan) [inline projects/storefrontlib/src/cms-structure/page/slot/page-slot.component.ts]: Do we need `?` here?

File: projects/storefrontlib/src/cms-structure/page/slot/page-slot.component.ts [hunk 3]
```diff
@@ -14,145 +14,184 @@ import {
...
+    return this.pendingComponentCount;
+  }
+
+  /**
+   * Toggles the css classes for the host element.
+   *
+   * @param cls the classname that should be added.
+   * @param forceAdd indicates that the class should be added or not regardless of the current classlist.
+   */
+  protected toggleStyleClass(cls: string, forceAdd?: boolean): void {
```
Comment (dunqan) [inline projects/storefrontlib/src/cms-structure/page/slot/page-slot.component.ts]: Did you plan to reuse this logic?

File: projects/storefrontlib/src/cms-structure/page/slot/page-slot.component.ts [hunk 4]
```diff
@@ -14,145 +14,184 @@ import {
...
   ngOnInit() {
     this.subscription.add(
-      this.components$.subscribe((components) => {
-        this.hasComponents = components && components.length > 0;
-        this.pendingComponentCount = components ? components.length : 0;
-        this.isPending = this.pendingComponentCount > 0;
-      })
+      this.slot$
+        .pipe(tap((slot) => this.decorate(slot)))
+        .subscribe((value) => (this.components = value?.components || []))
```
Comment (dunqan) [inline projects/storefrontlib/src/cms-structure/page/slot/page-slot.component.ts]: As we don't have async pipe anywhere inside, I'm wondering if addin cd.markForCheck() here won't be needed in some cases?

File: projects/storefrontlib/src/cms-structure/page/slot/page-slot.component.ts (hunk 3 above)
Comment (tobi-or-not-tobi) [reply projects/storefrontlib/src/cms-structure/page/slot/page-slot.component.ts:142]: I thought yes, but indeed it's not needed as the BehaviourSubject is initialised explicitly.

File: projects/storefrontlib/src/cms-structure/page/slot/page-slot.component.ts (hunk 2 above)
Comment (tobi-or-not-tobi) [reply projects/storefrontlib/src/cms-structure/page/slot/page-slot.component.ts]: I thought yes, but indeed it's not needed as the BehaviourSubject is initialised explicitly.

File: projects/storefrontlib/src/cms-structure/page/slot/page-slot.component.ts (hunk 3 above)
Comment (tobi-or-not-tobi) [reply projects/storefrontlib/src/cms-structure/page/slot/page-slot.component.ts]: no! left over. thx.

File: projects/storefrontlib/src/cms-structure/page/slot/page-slot.component.ts (hunk 4 above)
Comment (tobi-or-not-tobi) [reply projects/storefrontlib/src/cms-structure/page/slot/page-slot.component.ts]: Good point. Not happing in our code base, but could happen if the page slot component is used in a situation where the slot data has not been loaded upfront.

**Review by dunqan — DISMISSED** [review]

**Review by dunqan — APPROVED** [review]

**Pull request merged**